containers/src/Kokkos_Functional.hpp

```diff
@@ -46,16 +46,29 @@
 #include <Kokkos_Macros.hpp>
 #include <impl/Kokkos_Functional_impl.hpp>
 
+#if defined KOKKKOS_ENABLE_CXX17
+#define KOKKOS_DEPRECATED_WITH_COMMENT_17(comment) \
+  KOKKOS_DEPRECATED_WITH_COMMENT(comment)
+#else
+#define KOKKOS_DEPRECATED_WITH_COMMENT_17(comment)
+#endif
+
 namespace Kokkos {
 
 // These should work for most types
 
 template <typename T>
 struct pod_hash {
-  using argument_type        = T;
-  using first_argument_type  = T;
-  using second_argument_type = uint32_t;
-  using result_type          = uint32_t;
+#if defined KOKKOS_ENABLE_CXX14 || defined KOKKOS_ENABLE_CXX17
+  using argument_type KOKKOS_DEPRECATED_WITH_COMMENT_17(
+      "(deprecated in C++17)(removed in C++20)") = T;
```

Comment:
I don't get the deprecation message

Reply:
After some discussion with Damien I propose to drop the message completely. Hinting towards the cause (deprecation of [binary_function](https://en.cppreference.com/w/cpp/utility/functional/binary_function) in the std) is kind of misleading and [this](https://godbolt.org/z/j1zE6fq5W) indicates that compilers are not rushing to remove it.